presto-hive/src/test/java/com/facebook/presto/hive/parquet/AbstractTestParquetReader.java

```diff
@@ -448,17 +776,71 @@ protected Double computeNext(Double previous)
 
     private static ContiguousSet<Integer> intsBetween(int lowerInclusive, int upperExclusive)
     {
-        return ContiguousSet.create(Range.openClosed(lowerInclusive, upperExclusive), DiscreteDomain.integers());
+        return ContiguousSet.create(Range.closedOpen(lowerInclusive, upperExclusive), DiscreteDomain.integers());
     }
 
     private static ContiguousSet<Long> longsBetween(long lowerInclusive, long upperExclusive)
     {
-        return ContiguousSet.create(Range.openClosed(lowerInclusive, upperExclusive), DiscreteDomain.longs());
+        return ContiguousSet.create(Range.closedOpen(lowerInclusive, upperExclusive), DiscreteDomain.longs());
     }
 
     private static ContiguousSet<BigInteger> bigIntegersBetween(BigInteger lowerInclusive, BigInteger upperExclusive)
     {
-        return ContiguousSet.create(Range.openClosed(lowerInclusive, upperExclusive), DiscreteDomain.bigIntegers());
+        return ContiguousSet.create(Range.closedOpen(lowerInclusive, upperExclusive), DiscreteDomain.bigIntegers());
+    }
+
+    private <F, S> List<List> createTestStruct(Iterable<F> firstParamValues, Iterable<S> secondParamValues)
+    {
+        List<List> result = new ArrayList<>();
+        Iterator<F> firstParam = firstParamValues.iterator();
+        Iterator<S> secondParam = secondParamValues.iterator();
+        while (firstParam.hasNext()) {
+            result.add(Arrays.asList(firstParam.next(), secondParam.next()));
+        }
+        return result;
+    }
+
+    private <F, S, T> List<List> createTestStruct(Iterable<F> firstParamValues, Iterable<S> secondParamValues, Iterable<T> thirdParamValues)
+    {
+        List<List> result = new ArrayList<>();
+        Iterator<F> firstParam = firstParamValues.iterator();
+        Iterator<S> secondParam = secondParamValues.iterator();
+        Iterator<T> thirdParam = thirdParamValues.iterator();
+        while (firstParam.hasNext() && secondParam.hasNext() && thirdParam.hasNext()) {
```

Comment:
Do you expect `firstParam`, `secondParam` and `thirdParam` to have a next element (`hasNext()` returns true for all)? If so, please add a `checkArgument` to assert that, otherwise you should handle that case.

Reply:
done